docs/extend/extend-zowe-overview.md

```diff
@@ -1,56 +1,52 @@
 # Extending Zowe
 
-Zowe was designed to be an extensible tools platform. You can extend it in several ways to meet your needs or distribute the plug-ins to users who have already installed Zowe and want to introduce new functionality to it. 
+Zowe is designed as an extensible tools platform. One of the Zowe architecture goals is to provide consistent interoperability between all Zowe components including extensions. The Zowe Conformance Program defines the criteria to help accomplish the aforementioned goal. By satisfying the Zowe Conformance Program criteria, extension providers are assured that their software remains functional throughout the Zowe release cycle. For more information, see the [Zowe Conformance Program](zowe-conformance-program.md).
 
-One of the goals of Zowe is to give users a consistent user experience, common functionality, and interoperability when using Zowe that includes the base set of Zowe core functions and plug-ins that are built outside the Zowe community. The Zowe Conformance Program provides a set of criteria to help with this. When followed, it also gives plug-in providers confidence that their software remains functional through Zowe releases. For more information, see [Zowe Conformance Program](zowe-conformance-program.md).
+Zowe can be extended in the following ways:
 
-You can extend Zowe in the following ways:
+- [Extend Zowe CLI](#extend-zowe-cli)
+- [Extend Zowe API Mediation Layer](#extend-zowe-api-mediation-layer)
+  - [Dynamic API registration](#dynamic-api-registration)
```

Comment:
Maybe we should keep this list flat, i.e. not enumerate the ways to extend API ML, because soon there will be more ways to extend API ML and they don't fit the level of onboarding new services.

Reply:
I think we should keep it as is and update this article if additional ways of extending zowe are added.